scipy/special/_basic.py

```diff
@@ -2445,6 +2520,10 @@ def factorial2(n, exact=False):
         Double factorial of `n`, as an int or a float depending on
         `exact`.
 
+    References
+    ----------
+    ..[1] https://functions.wolfram.com/GammaBetaErf/Factorial2/27/01/0002/
```

Comment:
These links constantly change. If it is possible to refer to a book or an article then mark it with a DOI

    ..[1] Some reference, BLA press, 1973, :doi:`10XXXX.XXXXXX`

Reply:
I realized that this function was basically just pasting together the function from [here](https://en.wikipedia.org/wiki/Double_factorial#Complex_arguments), i.e.
```
n!! = 2 ** n * sqrt(2/pi) * Γ(n / 2 + 1)   # for n odd integer
```
with the trivially true
```
n!! = 2 ** n * Γ(n / 2 +1)  # for n even integer
```

AFAICT, the choice of modulation term `(pi / 2) ** ((cos(pi * z) - 1)/4)` is completely arbitrary and just a "nice" choice to get an exponent that does the right thing at even/odd integers.

I feel even less excited about using such an arbitrary construction than accepting the divergence of sqrt(2/pi) = 0.798 at even integers between the canonical extension and the integer-only result. However, I think the mechanism of explicitly opting in to `extend="complex"` is good enough for still doing so, and we could couple it with a big enough warning in the docs.

WDYT? @ilayn @j-bowhay

Reply:
> however I think the complex extension is still useful.

Yeah, I agree, but I'd like to stay with the _canonical_ extension. The surprising thing is that the extension does not match for even integers, but I think that behaviour change think is "opt-in enough" through the keyword, and can be documented with a big note in the docstring.

Reply:
@ilayn 
WDYT about using the canonical extension for the double-factorial, even if it doesn't match on even integers (and adding a big note to the docstring about that)?

Reply:
> @ilayn
> WDYT about using the canonical extension for the double-factorial, even if it doesn't match on even integers (and adding a big note to the docstring about that)?

Gentle ping @ilayn

CC @tupui

Reply:
Sorry complex numbers is really not my area.

Reply:
Since having the factorial fixed to be consistent with each other / comb etc is probably the more important fix maybe it would be worth pushing back any extension to the complex domain to a follow up pr making this easier to review / merge?

Reply:
Resolving this thread as the reference has been removed, along with the ambiguous extension.